src/sentry/rules/conditions/event_attribute.py

```diff
@@ -269,10 +133,14 @@ def _passes(self, attribute_values: Sequence[object | None]) -> bool:
 
     def passes(self, event: GroupEvent, state: EventState, **kwargs: Any) -> bool:
         attr = self.get_option("attribute", "")
-        try:
-            attribute_values = self._get_attribute_values(event, attr.lower())
-        except KeyError:
+        path = attr.split(".")
+
+        first_attr = path[0]
+        attr_processor = attribute_registry.get(first_attr)
+        if not attr_processor:
             attribute_values = []
+        else:
+            attribute_values = attr_processor().handle(path, event)
```

Comment:
this is so much nicer

Reply:
this code is... [SOLID](https://en.wikipedia.org/wiki/SOLID) 🎉